ui/src/app/base/types.ts

```diff
@@ -104,12 +104,14 @@ export type ResourcePoolState = {
   saving: boolean;
 };
 
+type TestResult = -1 | 0 | 1;
+
 export type TestStatus = {
-  status: number;
-  pending: number;
-  running: number;
-  passed: number;
-  failed: number;
+  status: TestResult;
+  pending: TestResult;
+  running: TestResult;
+  passed: TestResult;
+  failed: TestResult;
```

Comment:
I don't think this is right. `status` can be any one of the following, according to `app/base/enum.js`:
``` js
export const scriptStatus = {
  NONE: -1,
  PENDING: 0,
  RUNNING: 1,
  PASSED: 2,
  FAILED: 3,
  TIMEDOUT: 4,
  ABORTED: 5,
  DEGRADED: 6,
  INSTALLING: 7,
  FAILED_INSTALLING: 8,
  SKIPPED: 9,
  APPLYING_NETCONF: 10,
  FAILED_APPLYING_NETCONF: 11,
};
```

And the rest of the parameters should just be `number` because they're simply counts of how many tests are in `pending`/`running`/etc state

Reply:
Ah quite right, thanks @Caleb-Ellis. There's no way to nicely represent a range in typescript, so I might just revert this and keep them all numbers.